test/counter.go

```diff
@@ -17,54 +17,37 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-type BaseValue func(*toolchainv1alpha1.ToolchainStatus)
-
-func MasterUserRecords(number int) BaseValue {
-	return func(status *toolchainv1alpha1.ToolchainStatus) {
-		status.Status.HostOperator = &toolchainv1alpha1.HostOperatorStatus{
-			MasterUserRecordCount: number,
-		}
-	}
+type CounterAssertion struct {
+	t      test.T
+	counts counter.Counts
 }
 
-func UserAccountsForCluster(clusterName string, number int) BaseValue {
-	return func(status *toolchainv1alpha1.ToolchainStatus) {
-		status.Status.Members = append(status.Status.Members, v1alpha1.Member{
-			ClusterName:      clusterName,
-			UserAccountCount: number,
-		})
+func AssertThatCounters(t *testing.T) *CounterAssertion {
+	counts, err := counter.GetCounts()
+	require.NoError(t, err)
+	return &CounterAssertion{
+		t:      t,
+		counts: counts,
 	}
 }
 
-func AssertThatUninitializedCounterHas(t *testing.T, baseValues ...BaseValue) {
+func AssertThatUnitializedCounters(t *testing.T) *CounterAssertion {
 	counts, err := counter.GetCounts()
-	assert.EqualErrorf(t, err, "counter is not initialized", "should be error because counter hasn't been initialized yet")
-	verifyCountsAndMetrics(t, counts, baseValues...)
+	require.EqualErrorf(t, err, "counter is not initialized", "should be error because counter hasn't been initialized yet")
+	return &CounterAssertion{
+		t:      t,
+		counts: counts,
+	}
 }
 
-func AssertThatCounterHas(t *testing.T, baseValues ...BaseValue) {
-	counts, err := counter.GetCounts()
-	require.NoError(t, err)
-	verifyCountsAndMetrics(t, counts, baseValues...)
+func (a *CounterAssertion) HaveMasterUserRecords(number int) *CounterAssertion {
+	assert.Equal(a.t, number, a.counts.MasterUserRecordCount)
+	return a
 }
 
-func verifyCountsAndMetrics(t *testing.T, counts counter.Counts, baseValues ...BaseValue) {
-
-	toolchainStatus := &v1alpha1.ToolchainStatus{
-		Status: v1alpha1.ToolchainStatusStatus{},
-	}
-	for _, apply := range baseValues {
-		apply(toolchainStatus)
-	}
-	require.NotNil(t, toolchainStatus.Status.HostOperator)
-	assert.Equal(t, toolchainStatus.Status.HostOperator.MasterUserRecordCount, counts.MasterUserRecordCount)
-	AssertMetricsGaugeEquals(t, toolchainStatus.Status.HostOperator.MasterUserRecordCount, metrics.MasterUserRecordGauge)
-
-	assert.Len(t, counts.UserAccountsPerClusterCounts, len(toolchainStatus.Status.Members))
-	for _, member := range toolchainStatus.Status.Members {
-		assert.Equal(t, member.UserAccountCount, counts.UserAccountsPerClusterCounts[member.ClusterName])
-		AssertMetricsGaugeEquals(t, member.UserAccountCount, metrics.UserAccountGaugeVec.WithLabelValues(member.ClusterName))
```

Comment:
could you please point me to the place in the new code where you verify the gauge metrics?

Reply:
oops, good catch! I was too eager when removing the `BaseValue`-related stuff

Reply:
assertions restored in f71a52d